mobilecoind/clients/python/blockchain_explorer.py

```diff
@@ -8,65 +8,79 @@
-        required=False,
-        default=5000,
-        help='Block Explorer listen port'
-    )
-    parser.add_argument(
-        '--mobilecoind_port',
-        type=int,
-        required=False,
-        default=4444,
-        help='Port of mobilecoind service to connect to'
-    )
-    parser.add_argument(
-        '--mobilecoind_host',
-        type=str,
-        required=False,
-        default='localhost',
-        help='Hostname of mobilecoind service to connect to'
-    )
+    parser.add_argument('--port',
+                        type=int,
+                        required=False,
+                        default=5000,
+                        help='Block Explorer listen port')
+    parser.add_argument('--mobilecoind_port',
+                        type=int,
+                        required=False,
+                        default=4444,
+                        help='Port of mobilecoind service to connect to')
+    parser.add_argument('--mobilecoind_host',
+                        type=str,
+                        required=False,
+                        default='localhost',
+                        help='Hostname of mobilecoind service to connect to')
     return parser.parse_args()
 
+
 @app.route('/')
 def index():
     num_blocks, num_transactions = client.get_ledger_info()
 
     blocks = []
+    signers = {}
     for i in range(num_blocks - 1, max(num_blocks - 100, -1), -1):
         _key_image_count, txo_count = client.get_block_info(i)
         # Will get ResourceExhausted if message larger than 4194304
         if txo_count > 20000:
             continue
         block = client.get_block(i)
-        size_of_block = getsizeof(block) * .001
-        block_row = (i, txo_count, size_of_block, bytes.hex(block.block.contents_hash.data))
+        block_row = (i, txo_count, bytes.hex(block.block.contents_hash.data),
+                     len(block.signatures))
         blocks.append(block_row)
 
+        # Process signature data - sort by signer
+        for signature_data in block.signatures:
+            signature = signature_data.signature.signature
+            signer = bytes.hex(signature_data.signature.signer.data)
+            # If a new signer has appeared, prepend False for all previous blocks
+            if signer not in signers:
+                signers[signer] = [False for i in range(i - 1)]
+            signers[signer].append(True)
+
     return render_template('index.html',
-        blocks=blocks,
-        num_blocks=num_blocks,
-        num_transactions=num_transactions)
+                           blocks=blocks,
+                           num_blocks=num_blocks,
+                           num_transactions=num_transactions,
+                           signers=signers)
+
 
 @app.route('/block/<block_num>')
 def block(block_num):
     block = client.get_block(int(block_num))
     size_of_block = getsizeof(block)
+
+    for signature in block.signatures:
+        signature.src_url = signature.src_url.split('/')[-2]
```

Comment:
This might come and bite us in the future, for partners who don't have the node name be a part of the URL.

Reply:
Agreed - for now they are matching this convention, and we can always address this with different heuristics as we need.